server/bundler/loader.js

```diff
@@ -66,6 +66,7 @@ function splitTemplate( path, module, chunkName ) {
 
 	result = [
 		'page( ' + path + ', function( context, next ) {',
+		'	context.sectionRouteMatched = true;',
```

Comment:
When you mentioned `/stats/123/12/4` did you mean that it first visibly loads the site selection and only then it loads the 404? Is there a general solution to this problem?